### DISCUSSION TITLE
Questions about makeAutoObservable

### DISCUSSION THREAD

**Discussion by devuxer** [mobxjs/mobx#3305] [Q&A] [answered]
1. Does `makeAutoObservable` ignore readonly fields?
2. If it does not, why?
3. How do I verify for myself how `makeAutoObservable` has annotated each field, property, getter, setter, method, etc. in a class?

**Comment by kubk** [comment, marked as answer]
Do you mean TS readonly types? No it doesnt, because TS readonly types don't exist in runtime. `makeAutoObservable` is all about automatic annotations, if you need more precise control you can use [makeObservable](https://mobx.js.org/observable-state.html#makeobservable). Also this ESLint plugin can be helpful for you: https://github.com/mobxjs/mobx/tree/main/packages/eslint-plugin-mobx#mobxexhaustive-make-observable

**Reply by devuxer** [reply]
@kubk,

I've been using TypeScript so long, I sometimes forget which features are TS vs JS! Anyway, that all makes sense, and thanks for the tip about the linter.

As for "more precise control", from the docs, it looks like you can gain pretty precise control even with `makeAutoObservable` by overriding only the annotations you disagree with (in my case, I could override just the readonly fields, settings them to `false`). Am I understanding that correctly?

**Reply by kubk** [reply]
@devuxer Yes, you can exclude readonly fields manually. I also prefer `makeAutoObservable` because it's less verbose 👍
